src/test/java/spoon/test/ctCase/CtCaseImplTest.java

```diff
@@ -0,0 +1,58 @@
+package spoon.test.ctCase;
+
+import org.junit.jupiter.api.Test;
+import spoon.Launcher;
+import spoon.reflect.code.CtCase;
+import spoon.reflect.code.CtStatement;
+import spoon.reflect.code.CtStatementList;
+import spoon.reflect.declaration.CtClass;
+import spoon.reflect.visitor.filter.TypeFilter;
+import spoon.support.compiler.FileSystemFile;
+import spoon.support.reflect.code.CtStatementListImpl;
+import spoon.test.ctCase.testclasses.ClassWithSwitchExample;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CtCaseImplTest {
+
+    @Test
+    void testInsertBeginWithListOfStatements() {
+        // contract: insertBegin adds a list of statements at the beginning of a case, i.e the list is added above two
+        // exiting statements, and the order of statements in the list remains the same
+
+        // arrange
+        CtClass classWithSwitchExample = getClassWithSwitchExample();
+        CtStatement firstStatementToBeInserted = createStatement(classWithSwitchExample, "firstStatement");
+        CtStatement secondStatementToBeInserted = createStatement(classWithSwitchExample, "secondStatement");
+        CtCase testCase = classWithSwitchExample.getElements(new TypeFilter<>(CtCase.class)).get(0);
+
+        CtStatementList ctStatementList =  new CtStatementListImpl<CtStatement>() {
+            {
+                addStatement(firstStatementToBeInserted);
+                addStatement(secondStatementToBeInserted);
+            }
+        };
```

Comment:
Don't create anonymous subtypes unless it's actually part of the test. Here, just use the factory.

```suggestion
        CtStatementList statementList = factory.createStatementList();
        statementList.addStatement(firstStatementToBeInserted);
        statementList.addStatement(secondStatementToBeInserted);
```

Reply:
I actually intended to initialize the list with those statements that's why created that way

Reply:
I like [this answer](https://stackoverflow.com/a/27521360) by Lukas Eder. Double brace init can be the cause of subtile bugs and isn't really buying you much. The memory downsides don't really apply here as it is in a test but seeing it in real code always tingles my spidey-senses.